gluten-data/src/main/java/org/apache/gluten/vectorized/ColumnarBatchOutIterator.java

```diff
@@ -80,7 +80,11 @@ public IMetrics getMetricsInternal() throws IOException, ClassNotFoundException
   }
 
   public long spill(long size) {
-    return nativeSpill(iterHandle, size);
+    if (!closed.get()) {
```

Comment:
Recently I also meet this issue(we haven't include this pr yet). But I'm curious in this way, IIUC, we can't spill memory that this `NativeMemoryManager` holding if the iterator is closed? Since we still hold the memory after the iterator is closed.

If it's an issue, can we directly get `NativeMemoryManager` to spill itself? By calling `shrinkCapacity({pool}, remaining)` in the native.

@zhztheplayer @WangGuangxin